File: app/src/main/java/social/entourage/android/authentication/AuthenticationModule.java
package social.entourage.android.authentication;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Module related to Application
 * Providing Authentication related dependencies
 */
@Module
public class AuthenticationModule {

    @Provides
    @Singleton
    public AuthenticationController providesAuthenticationController(ComplexPreferences userSharedPref) {
        return new AuthenticationController(userSharedPref).init();
    }

    @Provides
    @Singleton
    public ComplexPreferences providesUserSharedPreferences(Application application) {
        return ComplexPreferences.getComplexPreferences(application, "userPref", Context.MODE_PRIVATE);
    }
}

File: app/src/main/java/social/entourage/android/authentication/AuthenticationModule.kt
package social.entourage.android.authentication

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import social.entourage.android.authentication.ComplexPreferences.Companion.getComplexPreferences
import javax.inject.Singleton

/**
 * Module related to Application
 * Providing Authentication related dependencies
 */
@Module
class AuthenticationModule {
    @Provides
    @Singleton
    fun providesAuthenticationController(userSharedPref: ComplexPreferences?): AuthenticationController {
        return AuthenticationController(userSharedPref!!).init()
    }

    @Provides
    @Singleton
    fun providesUserSharedPreferences(application: Application?): ComplexPreferences {
        return getComplexPreferences(application!!, "userPref", Context.MODE_PRIVATE)
    }
}